### ISSUE TITLE
Remove need for FPM in order to have Portals

### ISSUE THREAD

**Issue by digitalextremist** [espocrm/espocrm#2740]
The @tmachyshyn implementation via `docker-compose.yml` does not use `FPM` which then breaks the portal implementation:

* `docker-compose.yml`: https://docs.espocrm.com/administration/docker/installation/#install-espocrm-with-docker-compose
* `Portal nginx.conf`: https://docs.espocrm.com/administration/portal/nginx-configuration/

This was how I resolved it so far:

```php
//de HACK 
if (function_exists('apache_request_headers') ) {
        $headers = apache_request_headers();
        if (array_key_exists("Espo-Portal-ID", $headers)) {
                $_SERVER['ESPO_PORTAL_ID'] = apache_request_headers()['Espo-Portal-ID'];
                if ($_SERVER['ESPO_PORTAL_ID']) {
                        $_SERVER['ESPO_PORTAL_IS_CUSTOM_URL'] = true;
                }
        }
}
```

Right at the top of `public/portal/index.php` as in my fork:

https://github.com/MigratorONE/espocrm/commit/50e735a11cfb748654d5d840ca76461462c68123

Wondering where I ought to put that, or how best to do that, then update the documentation to include:

```
proxy_set_header Espo-Portal-ID $espocrm_portal;
```

Or similar: overall the `docker-compose` documentation has separate from the `FPM` implementation which the rest of the documentation for `EspoCRM` itself seems to use exclusively.

**Comment by digitalextremist** [comment]
First condition testing for function is so that `WebSocket` does not fail out or error to the log constantly.

**Comment by yurikuzn** [comment]
I don't think this should be 'fixed' in codebase. Espo just requires 1 or 2 env variables which should be not problem to set in the container configuration.

**Comment by digitalextremist** [comment]
Commented in @yurikuzn: https://github.com/espocrm/espocrm/pull/2741#issuecomment-1546970356

My only concern personally is to have a `docker-compose.yml` implementation which is not based on `fpm` unless there is some huge performance benefit, or other benefit. The @tmachyshyn `docker-compose.yml` plus a reverse proxy does not allow for custom portal domains without some measure being taken, and added to the documentation.

I have a working implementation now, with reverse proxy. I do not mind contributing _something_ and understand the need to keep hacks out. I just want something to stick that avoids the problem for others, and replicates well in future instances easily.
